tests/test_optimizer_random_dag.py

```diff
@@ -65,8 +64,8 @@ def generate_random_dag(
                 sky.Resources(
                     cloud=CLOUDS[candidate.cloud],
                     instance_type=candidate.instance_type \
-                        if candidate.cloud != 'GCP' \
-                        else random.choice(GCP_INSTANCE_TYPES),
+                        if not pd.isna(candidate.instance_type) \
+                        else CLOUDS['GCP'].get_default_instance_type(),
```

Comment:
Remove backslashes and quote in parens so yapf can auto-reformat?

Reply:
Fixed. For readability, I removed the set comprehension and added a loop statement instead.